proto/zitadel/system.proto

```diff
@@ -0,0 +1,611 @@
+                description: "details of ZITADEL's event driven software concepts";
+            };
+            responses: {
+                key: "200";
+                value: {
+                    description: "Events which were not processed by the views";
+                };
+            };
+        };
+    }
+
+    //Deletes the event from failed events view.
+    // the event is not removed from the change stream
+    // This call is usefull if the system was able to process the event later. 
+    // e.g. if the second try of sending an email was successful. the first try produced a
+    // failed event. You can find out if it worked on the `failure_count`
+    rpc RemoveFailedEvent(RemoveFailedEventRequest) returns (RemoveFailedEventResponse) {
+        option (google.api.http) = {
+            delete: "/failedevents/{database}/{view_name}/{failed_sequence}";
+        };
+
+        option (zitadel.v1.auth_option) = {
+            permission: "iam.write";
+        };
+
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "failed events";
+            external_docs: {
+                url: "https://docs.zitadel.ch/concepts#Software_Architecture";
+                description: "details of ZITADEL's event driven software concepts";
+            };
+            responses: {
+                key: "200";
+                value: {
+                    description: "Events removed from the list";
+                };
+            };
+            responses: {
+                key: "400";
+                value: {
+                    description: "failed event not found";
+                    schema: {
+                        json_schema: {
+                            ref: "#/definitions/rpcStatus";
+                        };
+                    };
+                };
+            };
+        };
+    }
+}
+
+
+//This is an empty request
+message HealthzRequest {}
+
+//This is an empty response
+message HealthzResponse {}
+
+message ListInstancesRequest {
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
+		json_schema: {
+			description: "Search query for lists";
+			required: ["query"]
+		};
+	};
+
+    //list limitations and ordering
+    zitadel.v1.ListQuery query = 1;
+    // the field the result is sorted
+    zitadel.instance.v1.InstanceFieldName sorting_column = 2;
+    //criterias the client is looking for
+    repeated zitadel.instance.v1.OrgQuery queries = 3;
+}
+
+message ListInstancesResponse {
+    zitadel.v1.ListDetails details = 1;
+    zitadel.instance.v1.OrgFieldName sorting_column = 2;
+    repeated zitadel.instance.v1.Instance result = 3;
+}
+
+message GetInstanceRequest {
+    string domain = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
+}
+
+message GetInstanceResponse {
+    zitadel.instance.v1.Instance instance = 1;
+}
+
+message AddInstanceRequest {
+    string domain = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
```

Comment:
is this meant as the custom domain? if so then there should not be min length and i would not have it as first field

Reply:
do we generate the default domain in metadel?